internal/controller/changetransferpolicy_controller.go

```diff
@@ -132,8 +138,25 @@ func (r *ChangeTransferPolicyReconciler) Reconcile(ctx context.Context, req ctrl
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ChangeTransferPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &promoterv1alpha1.ChangeTransferPolicy{}, ".status.proposed.hydrated.sha", func(rawObj client.Object) []string {
+		//nolint:forcetypeassert
+		ctp := rawObj.(*promoterv1alpha1.ChangeTransferPolicy)
+		return []string{ctp.Status.Proposed.Hydrated.Sha}
+	}); err != nil {
+		return fmt.Errorf("failed to set field index for .status.proposed.hydrated.sha: %w", err)
+	}
+
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &promoterv1alpha1.ChangeTransferPolicy{}, ".status.active.hydrated.sha", func(rawObj client.Object) []string {
+		//nolint:forcetypeassert
+		ctp := rawObj.(*promoterv1alpha1.ChangeTransferPolicy)
+		return []string{ctp.Status.Active.Hydrated.Sha}
+	}); err != nil {
+		return fmt.Errorf("failed to set field index for .status.active.hydrated.sha: %w", err)
+	}
```

Comment:
I want to setup all indexes for resources in the controller that manages the resource because it centralizes all that code and you can't have the same index field setup twice. 

So the rule is whatever resource you want an index on should be setup in the controller that manages that resource. Then every controller that uses the same client under the manager can use the index.

Reply:
Gotcha, then probably just need to add a comment to explain why the indexes exist.